cpp/src/strings/regex/regexec.cu

```diff
@@ -158,23 +157,26 @@ std::unique_ptr<reprog_device, std::function<void(reprog_device*)>>
     d_prog->_classes_count = classes_count;
     d_prog->_codepoint_flags = codepoint_flags;
     // allocate execute memory if needed
+    rmm::device_buffer* d_relists{};
     if( rlm_size > 0 )
     {
-        RMM_TRY(RMM_ALLOC(&(d_prog->_relists_mem),rlm_size,stream));
+        d_relists = new rmm::device_buffer(rlm_size, stream);
+        d_prog->_relists_mem = d_relists->data();
     }
 
     // copy flat prog to device memory
-    CUDA_TRY(cudaMemcpy(d_buffer,h_buffer.data(),memsize,cudaMemcpyHostToDevice));
+    CUDA_TRY(cudaMemcpy(d_buffer->data(),h_buffer.data(),memsize,cudaMemcpyHostToDevice));
     //
-    auto deleter = [](reprog_device*t) {t->destroy();};
+    auto deleter = [d_buffer, d_relists](reprog_device*t) {
+        t->destroy(); 
+        delete d_buffer;
```

Comment:
Is there any chance that this buffer's memory could be used on a different stream than it was created on before this deleter is called?

Reply:
I went through all uses of `reprog_device` and they are all limited to a single stream per instance.

Reply:
What do you think about updating https://github.com/vuule/cudf/blob/0115947b225262cc5502922366571c7a0ac64d51/cpp/src/strings/regex/regex.cuh#L76
to emphasize that the object should be used in the same stream it is created in?

Reply:
Good idea.

Reply:
Can you make that change in this PR?

Reply:
yeah, I assumed your vote expects this change.